### PR TITLE
chore: updates libs version to handle trivy vulnerabities

### PR DESCRIPTION
**Pull request by kotharironak**
## Description
Updates the different library version to take care of vulns

### Testing
Locally ran:
```
trivy image traceableai-docker.jfrog.io/hypertrace/raw-spans-grouper:test
trivy image traceableai-docker.jfrog.io/hypertrace/span-normalizer:test
```

will can merge after this : https://github.com/hypertrace/hypertrace-ingester/pull/395

### CODE REVIEW

File: hypertrace-ingester/build.gradle.kts [hunk 1]
```diff
@@ -26,12 +26,12 @@ hypertraceDocker {
 
 dependencies {
 
-  implementation("org.hypertrace.core.kafkastreams.framework:kafka-streams-framework:0.2.6")
-  implementation("org.hypertrace.core.serviceframework:platform-service-framework:0.1.49")
-  implementation("org.hypertrace.core.serviceframework:platform-metrics:0.1.49")
+  implementation("org.hypertrace.core.kafkastreams.framework:kafka-streams-framework:0.2.9")
```
Comment (aaron-steinfeld) [inline hypertrace-ingester/build.gradle.kts:29]: now bumped to `0.2.10`

Comment (kotharironak) [reply hypertrace-ingester/build.gradle.kts]: Done.

**Comment by aaron-steinfeld** [comment]
Lots of CVEs remaining here. A few dropped after we started upgrading the core libs so they got rev'd again, but a number are local to here to. Would start by updating your kafka streams, platform, grpc again, then continue from there.

**Comment by aaron-steinfeld** [comment]
not sure why build is not showing required, but still a number of things to fix here - lots of vulns, and code scan uploads are failing because they each need a distinct category

**Comment by kotharironak** [comment]
> not sure why build is not showing required, but still a number of things to fix here - lots of vulns, and code scan uploads are failing because they each need a distinct category

Working on this today. Was distracted due to participation in hack.

**Comment by kotharironak** [comment]
> Lots of CVEs remaining here. A few dropped after we started upgrading the core libs so they got rev'd again, but a number are local to here to. Would start by updating your kafka streams, platform, grpc again, then continue from there.

Can you relook now?

File: .github/workflows/pr-build.yml [hunk 2]
```diff
@@ -35,54 +35,63 @@ jobs:
         with:
           image: hypertrace/hypertrace-ingester
           output-mode: github
+          category: hypertrace-ingester
```
Comment (kotharironak) [inline .github/workflows/pr-build.yml:38]: Added to fix this issue: https://github.com/hypertrace/hypertrace-ingester/actions/runs/5452280343/jobs/9919573254?pr=396

Comment (kotharironak) [reply .github/workflows/pr-build.yml:38]: Raised separate a PR for it take an effect under `pull_request_traget` : https://github.com/hypertrace/hypertrace-ingester/pull/397

File: owasp-suppressions.xml [hunk 3]
```diff
@@ -1,3 +1,90 @@
...
+   ]]></notes>
+    <packageUrl regex="true">^pkg:maven/org\.hypertrace\.config\.service/span\-processing\-utils@.*$</packageUrl>
+    <cpe>cpe:/a:utils_project:utils</cpe>
+    <cpe>cpe:/a:processing:processing</cpe>
+  </suppress>
+  <suppress>
+    <notes><![CDATA[
+   file name: span-processing-config-service-api-0.1.52.jar
+   ]]></notes>
+    <packageUrl regex="true">^pkg:maven/org\.hypertrace\.config\.service/span\-processing\-config\-service\-api@.*$</packageUrl>
```
[the remark this answers is not in the document]
Comment (kotharironak) [reply owasp-suppressions.xml]: Done.

File: hypertrace-metrics-exporter/hypertrace-metrics-exporter/build.gradle.kts [hunk 4]
```diff
@@ -36,14 +36,19 @@ dependencies {
...
-  implementation("com.google.protobuf:protobuf-java:3.22.0")
+  implementation("com.google.protobuf:protobuf-java:3.23.3")
 
   // open telemetry proto
   implementation("io.opentelemetry:opentelemetry-proto:1.6.0-alpha")
 
   // kafka
-  implementation("org.apache.kafka:kafka-clients:7.2.1-ccs")
-
+  implementation("org.apache.kafka:kafka-clients:7.4.0-ccs")
```
[the remark this answers is not in the document]
Comment (kotharironak) [reply hypertrace-metrics-exporter/hypertrace-metrics-exporter/build.gradle.kts]: Done.

File: hypertrace-trace-enricher/enriched-span-constants/build.gradle.kts [hunk 5]
```diff
@@ -63,7 +63,8 @@ sourceSets {
 }
 
 dependencies {
-  api("com.google.protobuf:protobuf-java-util:3.23.3")
+  api(platform("org.hypertrace.core.kafkastreams.framework:kafka-bom:0.2.14"))
```
Comment (aaron-steinfeld) [inline hypertrace-trace-enricher/enriched-span-constants/build.gradle.kts:66]: I don't think we want the bom in the constants - they're not using kafka (I hope!) and that means we're dictating a kafka version to any consumer of the constants.

File: hypertrace-view-generator/hypertrace-view-generator-api/build.gradle.kts [hunk 6]
```diff
@@ -13,10 +13,6 @@ sourceSets {
...
 
 dependencies {
-  api("org.apache.avro:avro:1.11.1")
-  constraints {
-    api("com.fasterxml.jackson.core:jackson-databind:2.14.2") {
-      because("version 2.12.7.1 has a vulnerability https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-3038424")
-    }
-  }
+  api("org.apache.avro:avro")
+  api(platform("org.hypertrace.core.kafkastreams.framework:kafka-bom:0.2.14"))
```
Comment (aaron-steinfeld) [inline hypertrace-view-generator/hypertrace-view-generator-api/build.gradle.kts:17]: likewise here.

File: hypertrace-metrics-generator/hypertrace-metrics-generator-api/build.gradle.kts [hunk 7]
```diff
@@ -22,14 +22,9 @@ sourceSets {
 
 dependencies {
   implementation("com.google.protobuf:protobuf-java:3.23.3")
-  implementation("org.apache.kafka:kafka-clients:7.4.0-ccs")
+  implementation(platform("org.hypertrace.core.kafkastreams.framework:kafka-bom:0.2.14"))
```
Comment (aaron-steinfeld) [inline hypertrace-metrics-generator/hypertrace-metrics-generator-api/build.gradle.kts:25]: Here too, although this looks like somehow the api already depended on kafka clients so I guess not a regression

File: span-normalizer/raw-span-constants/build.gradle.kts [hunk 8]
```diff
@@ -57,7 +57,8 @@ sourceSets {
 }
 
 dependencies {
-  api("com.google.protobuf:protobuf-java-util:3.23.3")
+  api(platform("org.hypertrace.core.kafkastreams.framework:kafka-bom:0.2.14"))
```
Comment (aaron-steinfeld) [inline span-normalizer/raw-span-constants/build.gradle.kts:60]: and here

File: span-normalizer/span-normalizer-api/build.gradle.kts [hunk 9]
```diff
@@ -57,10 +57,6 @@ sourceSets {
...
 dependencies {
   api("com.google.api.grpc:proto-google-common-protos:2.14.1")
-  api("org.apache.avro:avro:1.11.1")
-  constraints {
-    api("com.fasterxml.jackson.core:jackson-databind:2.14.2") {
-      because("version 2.12.7.1 has a vulnerability https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-3038424")
-    }
-  }
+  api("org.apache.avro:avro")
+  api(platform("org.hypertrace.core.kafkastreams.framework:kafka-bom:0.2.14"))
```
Comment (aaron-steinfeld) [inline span-normalizer/span-normalizer-api/build.gradle.kts:61]: and here

File: hypertrace-trace-enricher/enriched-span-constants/build.gradle.kts (hunk 5 above)
Comment (kotharironak) [reply hypertrace-trace-enricher/enriched-span-constants/build.gradle.kts:66]: Reverting this.

Comment (kotharironak) [reply hypertrace-trace-enricher/enriched-span-constants/build.gradle.kts]: Done

File: hypertrace-view-generator/hypertrace-view-generator-api/build.gradle.kts (hunk 6 above)
Comment (kotharironak) [reply hypertrace-view-generator/hypertrace-view-generator-api/build.gradle.kts]: Done

File: hypertrace-metrics-generator/hypertrace-metrics-generator-api/build.gradle.kts (hunk 7 above)
Comment (kotharironak) [reply hypertrace-metrics-generator/hypertrace-metrics-generator-api/build.gradle.kts:25]: kept as is, as it is using kafka-client.

File: span-normalizer/raw-span-constants/build.gradle.kts (hunk 8 above)
Comment (kotharironak) [reply span-normalizer/raw-span-constants/build.gradle.kts]: Done

File: span-normalizer/span-normalizer-api/build.gradle.kts (hunk 9 above)
Comment (kotharironak) [reply span-normalizer/span-normalizer-api/build.gradle.kts]: Done.

File: hypertrace-trace-enricher/enriched-span-constants/build.gradle.kts (hunk 5 above)
Comment (kotharironak) [reply hypertrace-trace-enricher/enriched-span-constants/build.gradle.kts]: Done

**Pull request merged**